pkgs/development/python-modules/orange-widget-base/default.nix

```
@@ -0,0 +1,59 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+, pyqt5
+, pyqtwebengine
+, matplotlib
+, orange-canvas-core
+, pyqtgraph
+, typing-extensions
+, qt5
+, pytestCheckHook
+, pytest-qt
+}:
+
+buildPythonPackage rec {
+  pname = "orange-widget-base";
+  version = "4.21.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-43JeEdgEb6WS7PySW1x08fn02yl9Q/HwwKo6ojaqiH0=";
+  };
+
+  propagatedBuildInputs = [
+    pyqt5
+    pyqtwebengine
+    matplotlib
+    orange-canvas-core
+    pyqtgraph
+    typing-extensions
+  ];
+
+  pythonImportsCheck = [ "orangewidget" ];
+
+  preCheck = ''
+    export HOME=$(mktemp -d)
+    export QT_PLUGIN_PATH="${qt5.qtbase.bin}/${qt5.qtbase.qtPluginPrefix}"
+    export QT_QPA_PLATFORM_PLUGIN_PATH="${qt5.qtbase.bin}/lib/qt-${qt5.qtbase.version}/plugins";
+    export QT_QPA_PLATFORM=offscreen
+  '';
+
+  checkInputs = [
+    pytest-qt
+    pytestCheckHook
+  ];
+
+  disabledTestPaths = [
+    "orangewidget/report/tests/test_report.py"
+    "orangewidget/tests/test_widget.py"
+  ];
+
+
+  meta = {
+    description = "Implementation of the base OWBaseWidget class and utilities for use in Orange Canvas workflows";
+    homepage = "https://github.com/biolab/orange-widget-base";
+    license = [ lib.licenses.gpl3 ];
```

```suggestion
    license = lib.licenses.gpl3Plus;
```
https://github.com/biolab/orange-widget-base/blob/ebc5230289b3b6e27b0887dfe77439276f269cc4/setup.py#L44